cmake/onnxruntime_providers.cmake

```diff
@@ -289,6 +289,28 @@ if (onnxruntime_USE_CUDA)
   endif()
 endif()
 
+if (onnxruntime_USE_TENSORRT OR onnxruntime_USE_DNNL)
+  file(GLOB_RECURSE onnxruntime_providers_shared_cc_srcs CONFIGURE_DEPENDS
```

Comment:
>onnxruntime_providers_shared [](start = 20, length = 28)

Suggest a better name onnxruntime_providers_interface.

Reply:
Yeah, this is tricky as I was thinking of moving other things into this in the future. Right now it's small just for the allocator.